File: React/learning-objects/src/Dog.js
import Animal from "./Animal";

function Dog(name, breed) {
  Animal.call(this, name);
  this.breed = breed;
}

Dog.prototype = Object.create(Animal.prototype);
Dog.prototype.constructor = Dog;

Dog.prototype.bark = function () {
  return "Woof! from " + this.name;
};

export default Dog;

File: OOP/Lesson6/Polymorphism/Dog.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lesson6.Polymorphism {
	internal class Dog : Animal {
		public string Breed { get; set; }

		public override void Speak() {
			Console.WriteLine("Dog says 1: Woof Woof");
		}

		public new void Speak2() {
			Console.WriteLine("Dog says 2: Woof Woof");
		}
	}
}
